src/sentry/web/frontend/base.py

```diff
@@ -509,26 +511,58 @@ def needs_sso(self, request: Request, organization: Organization) -> bool:
             return True
         return False
 
+    def _lookup_orm_org(self) -> Organization | None:
+        """
+        Used by convert_args to convert the hybrid cloud safe active_organization object into an org ORM.
+        This should really only be used by the Region or Monolith silo modes -- calling this in a Control silo
+        endpoint or codepath will result in exceptions.
+        :return:
+        """
+        organization: Organization | None = None
+        if self.active_organization:
+            try:
+                organization = Organization.objects.get(id=self.active_organization.organization.id)
+            except Organization.DoesNotExist:
+                pass
+        return organization
+
     def convert_args(
         self, request: Request, organization_slug: str | None = None, *args: Any, **kwargs: Any
     ) -> tuple[tuple[Any, ...], dict[str, Any]]:
-        # TODO:  Extract separate view base classes based on control vs region / monolith,
-        # with distinct convert_args implementation.
-        if SiloMode.get_current_mode() == SiloMode.CONTROL:
-            assert self.active_organization is not None
-            kwargs["organization"] = self.active_organization.organization
-        else:
-            organization: Organization | None = None
-            if self.active_organization:
-                for org in Organization.objects.filter(id=self.active_organization.organization.id):
-                    organization = org
+        if "organization" not in kwargs:
+            kwargs["organization"] = self._lookup_orm_org()
 
-            kwargs["organization"] = organization
+        return args, kwargs
 
-        return (args, kwargs)
 
+class RegionSiloOrganizationView(OrganizationView):
+    """
+    A view which has direct ORM access to organization objects.  In practice, **only endpoints that exist in the
+    region silo should use this class**.  When All endpoints have been convert / tested against region silo compliance,
+    the base class (OrganizationView) will likely disappear and only either ControlSilo* or RegionSilo* classes will
+    remain.
+    """
+
+    def convert_args(
+        self, request: Any, organization_slug: str | None = None, *args: Any, **kwargs: Any
+    ) -> tuple[tuple[Any, ...], dict[str, Any]]:
+        if "organization" not in kwargs:
+            kwargs["organization"] = self._lookup_orm_org()
 
-class ProjectView(OrganizationView):
+        return args, kwargs
+
+
+class ControlSiloOrganizationView(OrganizationView):
+    def convert_args(
+        self, request: Any, *args: Any, **kwargs: Any
+    ) -> tuple[tuple[Any, ...], dict[str, Any]]:
```

Comment:
Side thought... although I have mixed feelings about this "convert args" pattern, do you think it's common enough to define a reusable custom type for `tuple[tuple[Any, ...], dict[str, Any]]` in our typing utility module?

Reply:
I can't say I'm a big fan either, but I don't see the number of these views increasing.  Most new stuff is react an json apis anywho.  I can't say if it's worth a stronger typing given that.